components/style/values/computed/mod.rs

```diff
@@ -494,6 +495,24 @@ impl SVGPaint {
 /// <length> | <percentage> | <number>
 pub type LengthOrPercentageOrNumber = Either<Number, LengthOrPercentage>;
 
+#[cfg(feature = "gecko")]
+impl ToAnimatedValue for LengthOrPercentageOrNumber {
```

Comment:
Thanks @nox,
Do you mean that we had better to introduce new value type?

If my understanding is correct, we allow setting the lenght and percentage and number on this properties. (i.e. we can specify number and 'px' and '%', and serialized result should 'px' or '%', because other browser will add unit information to serialized result. For detail, see https://bugzilla.mozilla.org/1379908.)

Length type allow setting the unit less length when parsing mode is `PARSING_MODE_ALLOW_UNITLESS_LENGTH`, we set this mode via gecko. So it seems to we don't allow setting the unit less length when css parsing.

I tried using the `LengthOrPercentage`, but I could't specify the unit less length in this case.

I think that we had better to introduce new value type like `SVGLengthOrPercentage` which allow setting the unit-less length and length and percentage in this case, if add new value type.


What do you think about this type?